Libraries/LibWeb/TrustedTypes/TrustedTypePolicyFactory.cpp

```diff
@@ -154,6 +155,12 @@ bool TrustedTypePolicyFactory::is_html(const JS::Value& value)
     return value.is_object() && is<TrustedHTML>(value.as_object()) && as<TrustedHTML>(value.as_object()).data_is_set();
 }
 
+// https://w3c.github.io/trusted-types/dist/spec/#dom-trustedtypepolicyfactory-isscript
+bool TrustedTypePolicyFactory::is_script(const JS::Value& value)
+{
+    return value.is_object() && is<TrustedScript>(value.as_object()) && as<TrustedScript>(value.as_object()).data_is_set();
```

Comment:
```
if (value.is_object()) {
   if (auto* trusted_script = as_if<TrustedScript>(value.as_object()); trusted_script && trusted_script->data_is_set())
       return true;
}
return false;
```

Avoid multiple redundant dynamic_cast expressions

Reply:
I'm also skeptical whether this "has data been set" thing makes any sense. The spec just wants you to make sure that the object really is that native type.  as in, "does it have the [[TrustedTypeData]] slot" (made-up spec language). I think just having the data as a string as it says should be fine.